File: packages/mirascript/src/vm/lib/global/sequence/repeat.ts
import type { VmConst } from '../../../types/index.js';
import { arrayLen, expectConst, expectNumber, VmLib } from '../../helpers.js';

export const repeat = VmLib(
    (data, times) => {
        expectConst('data', data, []);
        const n = arrayLen(expectNumber('times', times));
        const result: VmConst[] = [];
        result.length = n;
        result.fill(data);
        return result;
    },
    {
        summary: '创建一个包含重复元素的数组',
        params: {
            data: { type: 'any', description: '要重复的元素' },
            times: { type: 'number', description: '重复的次数，必须是非负整数' },
        },
        returns: { type: 'type(data)[]' },
        examples: ['repeat(0, 5) // [0, 0, 0, 0, 0]', 'repeat("a", 3) // ["a", "a", "a"]'],
    },
);

File: crates/python/mirascript/_vm/lib/vm_global/sequence/repeat.py
from ....types import Uninitialized, VmAny
from ..._helpers import _expect_const, _expect_number, _required
from ..._helpers_utils import _array_len


def repeat(data: VmAny = Uninitialized, times: VmAny = Uninitialized):
    _expect_const("data", data, [])
    _required("times", times, [])
    n = _array_len(_expect_number("times", times))

    result = []
    for _ in range(n):
        result.append(data)
    return result
